### ISSUE TITLE
Suggestion - Do Not Disturb / Block Numbers

### ISSUE THREAD

**Issue by enutpeanut** [InZidiuZ/op-framework-issue-tracker#2612]
### Assurance

I have read the `README.md`

### Summary

There are two features that I have placed into one suggestion because they're alternative solutions to the same problem.

Do Not Disturb
This would be a button in the smartphone menu which will put the phone into 'Do Not Disturb' mode. You will not be able to hear anybody calling you while it is enabled, nor will you receive any notifications for texts. Once you open the phone, however, you will be able to see if somebody is calling you and still be able to receive messages.

Block Numbers
This feature will mean you can block specific numbers from calling you. In the contacts/messages/phone app in the smartphone, there is an option to block a number. You can choose how long to block a number for (e.g. 10 minutes, 1 hour, 24 hours, forever etc). While a number is blocked, you will not be able to receive any phone calls or messages from them. In the contacts app, there will be a section for 'blocked numbers'. In there, you can unblock anyone you have blocked.
While trying to call somebody who has blocked you, it can either ring them like it normally would but they wouldn't see that you're calling, or it can tell you that 'this number has blocked you'.

### Reason

Spam calling is very annoying, and this is a very direct solution to it.
While in a chase (PD or crim), it would be nice to put the phone into 'Do Not Disturb' or 'Silent' mode so you don't get distracted.
If you are trying to find somebody but they have blocked you, it can spawn an interesting roleplay situation. You would have to get somebody else to call them, or try and find that person face-to-face.
In my opinion, it would be a nice change and I can't really think of any downsides to it in the roleplay.

I actually think an implementation of both features could be nice, if it is easy enough, as they both cater to different situations nicely.

**Comment by coalaura** [comment]
Closing in favor of #2447
